pkg/client/apiutil/apimachinery.go

```diff
@@ -27,10 +27,24 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/client-go/discovery"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/restmapper"
 )
 
+var (
+	protobufScheme = runtime.NewScheme()
```

Comment:
Actually one nit: Can we add a setter for this so projects that have additional apis that do support protobuf (metrics-server, all the openshift apis that are served via its aggregated apiserver) can add themselves here?

Reply:
I am also fine doing a quick follow-up for that and merging this PR as-is

Reply:
> Actually one nit: Can we add a setter for this so projects that have additional apis that do support protobuf (metrics-server, all the openshift apis that are served via its aggregated apiserver) can add themselves here?

Sure. How about provide a function in `pkg/client/apiutil/apimachinery.go`?

```golang
func AdditionalProtobufScheme(builder runtime.SchemeBuilder) error {
    return builder.AddToScheme(protobufScheme)
}
```

Reply:
Sounds good to me. Could you also add a small godoc comment explaining it?

Reply:
How about `AddToProtobufScheme` to make the experience familiar with the other AddToScheme methods?

Reply:
I have add this func and a RWMutex to avoid race modification of protobufScheme.
 PTAL @alvaroaleman @vincepri